src/Spatial/Euclidean/PolyLine2D.cs

```diff
@@ -0,0 +1,224 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace MathNet.Spatial.Euclidean
+{
+    /// <summary>
+    /// The PolyLine2D class represents a 2D curve in space made up of line segments joined end-to-end, and is 
+    /// stored as a sequential list of 2D points.
+    /// </summary>
+    public class PolyLine2D : IEnumerable<Point2D>
+    {
+        // Internal storage for the points
+        private List<Point2D> _points;
+
+        /// <summary>
+        /// Returns the number of points in the polyline
+        /// </summary>
+        public int Count => this._points.Count;
+
+        /// <summary>
+        /// Returns the length of the polyline as the sum of the length of the individual segments
+        /// </summary>
+        public double Length => this.GetPolyLineLength();
+
+        /// <summary>
+        /// Default constructor, creates an empty PolyLine2D
+        /// </summary>
+        public PolyLine2D() : this(Enumerable.Empty<Point2D>())
```

Comment:
Is this ever useful since there is no way to add anything to it later?
`public static readonly PolyLine2D Empty = new PolyLine2D(Enumerable.Empty<Point2D>())`
Is an alternative